src/app/components/project-summary/project-summary.component.ts

```diff
@@ -0,0 +1,77 @@
+import { Component, OnInit, Input } from '@angular/core';
+import { ResourceService } from 'src/app/services/resource/resource.service';
+import { ApiService } from 'src/app/services/api.service';
+import { AllocatedResourceType, Project } from 'src/app/types/types';
+
+@Component({
+  selector: 'app-project-summary',
+  templateUrl: './project-summary.component.html',
+  styleUrls: ['./project-summary.component.scss'],
+})
+export class ProjectSummaryComponent implements OnInit {
+  constructor(
+    protected resourceService: ResourceService,
+    private apiService: ApiService
+  ) {}
+
+  @Input() currentProject!: Project | undefined;
+  @Input() resourcesWithNames!: AllocatedResourceType[];
+
+  ngOnInit(): void {}
+
+  /*
+    The vendor details are dynamic. Each vendor has different set of attributes.
+    Vendor A can have Department, Super Department values.
+    Vendor B can have Cost Center, Department values.
+    
+    This function dynamically creates the vendor details in the DOM based on the attributes of the 
+    specific vendor.
+  */
+  cleanVendorDetails() {
+    if (this.currentProject !== undefined) {
```

Comment:
We need to dicsucss why we need this 2 function we not supposed to use java script with angular